torch/_overrides.py

```diff
@@ -1097,3 +1104,22 @@ def get_overridable_functions():
                 continue
             overridable_funcs[namespace].append(func)
     return overridable_funcs
+
+@functools.lru_cache(None)
+def get_methods():
+    import torch
+    overridable_funcs = get_overridable_functions()
+    methods = set(overridable_funcs[torch.Tensor])
+    return methods
+
+def is_method_property(func):
```

Comment:
This function's current location (`torch._overrides.is_method_property`) isn't going to be suitable as public API, but we can fix this later.

Reply:
I moved the imports into the functions to avoid circular imports, and then imported this function in `torch.__init__`. Isn't that sufficient?

Reply:
That doesn't look good. The name doesn't really make sense to an unsuspecting user (should be something like `is_tensormethod__or_property`), and more importantly: why must this be public at all? Things like `has_torch_function` also aren't public.

Reply:
There is a need for consumers of `__torch_function__` to treat methods/properties differently. Specifically:

1. `Tensor.add` may be passed into `__torch_function__`, even for non-subclasses. This is by design and there's a use-case for it. See https://github.com/pytorch/pytorch/pull/37091#issuecomment-667402832
2. `Tensor.add` only accepts an instance of `Tensor` as its first argument.
3. Some methods don't have `__module__`, especially those defined in C.

This makes it necessary for one to detect that something is a method.

Reply:
Then I think it's time for a public submodule, e.g. `torch.overrides`. There's also `get_overridable_functions` that should go there, now we document to get it from `_overrides` and that it's available but may change at any time (see https://pytorch.org/docs/master/notes/extending.html#testing-coverage-of-overrides-for-the-pytorch-api).

Can you add the use case example to the docstring? Documentation seems necessary for people to make sense of this.

Reply:
Yeah I agree with Ralf here on the other things we need to do to make this public API

Reply:
Should I do that with this PR?

Reply:
> Should I do that with this PR?

Yes please. It should be pretty quick to do.